cmd/internal/pluginutils/bypath.go

```diff
@@ -0,0 +1,78 @@
+// Copyright 2023 Intel Corporation. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package pluginutils
+
+import (
+	"os"
+	"path"
+	"strings"
+
+	pluginapi "k8s.io/kubelet/pkg/apis/deviceplugin/v1beta1"
+)
+
+// Returns a slice of by-path Mounts for a pci device with pciAddress.
+// by-path files are searched from the given bypathDir.
+// In the by-path dir, any files that start with "pci-<pci addr>" will be added to mounts.
+func BypathMountsForPci(pciAddress, bypathDir string) []pluginapi.Mount {
+	var mounts []pluginapi.Mount
+
+	if pciAddress == "" {
+		return nil
+	}
+
+	files, err := os.ReadDir(bypathDir)
+	if err != nil {
+		return nil
+	}
+
+	linkPrefix := "pci-" + pciAddress
+
+	for _, f := range files {
+		if strings.HasPrefix(f.Name(), linkPrefix) {
+			absPath := path.Join(bypathDir, f.Name())
+			mounts = append(mounts, pluginapi.Mount{
+				ContainerPath: absPath,
+				HostPath:      absPath,
+				ReadOnly:      true,
+			})
+		}
+	}
+
+	return mounts
+}
+
+// Returns the pci address for a drm card by reading the
+// symbolic link that the /sys/class/drm/cardX points to.
+// ../../devices/pci0000:00/0000:00:02.0/drm/card
+// -------------------------^^^^^^^^^^^^---------.
+func ReadPciAddressForCard(cardPath string) string {
+	linkPath, err := os.Readlink(cardPath)
+	if err != nil {
+		return ""
+	}
+
+	parts := strings.Split(linkPath, "/")
+	l := len(parts)
+
+	if l < 3 {
+		return ""
+	}
+
+	if parts[l-2] != "drm" {
+		return ""
+	}
+
+	return parts[l-3]
```

Comment:
if the card info was available this should work:
```suggestion
	return filepath.Base(strings.TrimSuffix(linkpath, filepath.Join("drm", cardName)))
```

Reply:
Yeah, that directly didn't work, but the concept worked.